src/ParaTest/Coverage/CoverageMerger.php

```diff
@@ -2,43 +2,80 @@
 
 namespace ParaTest\Coverage;
 
+use SebastianBergmann\CodeCoverage\CodeCoverage;
+
 class CoverageMerger
 {
     /**
-     * @var \PHP_CodeCoverage
+     * @var \PHP_CodeCoverage|CodeCoverage
      */
     private $coverage = null;
 
-    public function addCoverage(\PHP_CodeCoverage $coverage)
+    /**
+     * @param \PHP_CodeCoverage|CodeCoverage $coverage
+     */
+    private function addCoverage($coverage)
     {
-        if ($this->coverage == null) {
-            $this->setCoverage($coverage);
+        if (null === $this->coverage) {
+            $this->coverage = $coverage;
         } else {
-            $this->mergeCoverage($coverage);
+            $this->coverage->merge($coverage);
         }
     }
 
     /**
-     * @param \PHP_CodeCoverage $coverage
+     * Returns coverage object from file.
+     *
+     * @param \SplFileObject $coverageFile Coverage file.
+     *
+     * @return \PHP_CodeCoverage|CodeCoverage
      */
-    private function setCoverage(\PHP_CodeCoverage $coverage)
+    private function getCoverageObject(\SplFileObject $coverageFile)
     {
-        $this->coverage = unserialize(serialize($coverage));
+        if ('<?php' === $coverageFile->fread(5)) {
+            return include $coverageFile->getRealPath();
```

Comment:
Instead of loading whole coverage file content, only first line is checked for php tag - should reduce memory usage  and improve performance.